### PR TITLE
Drop Python 3.5

### PR DESCRIPTION
**Pull request by jrbourbeau**
This PR drops support for Python 3.5 and is a companion PR to https://github.com/dask/dask/pull/5528 in which we dropped Python 3.5 in `dask/dask`

### CODE REVIEW

**Comment by mrocklin** [comment]
I'm happy to defer to the group opinion here.  I suspect that you all have
thought things through well.

**Comment by jrbourbeau** [comment]
We're getting a couple of test failures on Appveyor I haven't seen before:

```
================================== FAILURES ===================================
____________________________ test_connection_args _____________________________
Traceback (most recent call last):
  File "C:\projects\distributed\distributed\tests\test_security.py", line 200, in test_connection_args
    assert len(tls_12_ciphers) == 1
AssertionError: assert 0 == 1
  -0
  +1
______________________________ test_listen_args _______________________________
Traceback (most recent call last):
  File "C:\projects\distributed\distributed\tests\test_security.py", line 253, in test_listen_args
    assert len(tls_12_ciphers) == 1
AssertionError: assert 0 == 1
  -0
  +1
```

These checks were previous being skipped on Windows which was only run with Python 3.5

https://github.com/dask/distributed/blob/9429ffe735e3767b427fd0b18aaa29d6a7da8513/distributed/tests/test_security.py#L197

@jcrist if you have a moment would you be able to take a look at this?

**Comment by mrocklin** [comment]
Given that @jcrist is out, I wonder if @mariusvniekerk can take a look at the TLS failures in Appveyor here:

```
================================== FAILURES ===================================
____________________________ test_connection_args _____________________________
Traceback (most recent call last):
  File "C:\projects\distributed\distributed\tests\test_security.py", line 203, in test_connection_args
    assert len(tls_12_ciphers) == 1
AssertionError: assert 0 == 1
  -0
  +1
---------------------------- Captured stderr call -----------------------------
[{'alg_bits': 128,
  'description': 'ECDHE-RSA-AES128-GCM-SHA256 TLSv1.2 Kx=ECDH     Au=RSA  '
                 'Enc=AESGCM(128) Mac=AEAD',
  'id': 50380847,
  'name': 'ECDHE-RSA-AES128-GCM-SHA256',
  'protocol': 'TLSv1/SSLv3',
  'strength_bits': 128}]
______________________________ test_listen_args _______________________________
Traceback (most recent call last):
  File "C:\projects\distributed\distributed\tests\test_security.py", line 259, in test_listen_args
    assert len(tls_12_ciphers) == 1
AssertionError: assert 0 == 1
  -0
  +1
---------------------------- Captured stderr call -----------------------------
[{'alg_bits': 128,
  'description': 'ECDHE-RSA-AES128-GCM-SHA256 TLSv1.2 Kx=ECDH     Au=RSA  '
                 'Enc=AESGCM(128) Mac=AEAD',
  'id': 50380847,
  'name': 'ECDHE-RSA-AES128-GCM-SHA256',
  'protocol': 'TLSv1/SSLv3',
  'strength_bits': 128}]
============================== warnings summary ===============================

```

**Comment by mrocklin** [comment]
OK, so what should we do here?  It seems like things are in an unpleasant state given that dask/dask has dropped 3.5, and so now our tests fail hard in some way.  We don't seem to be able to find someone to debug familiar enough to debug TLS issues.  Some options:

1.  We can revert the 3.5 drop in dask/dask
2.  We can live with the TLS failure, merge this, and hope that it's not significant
3.  We can reach out to others that might know more about TLS and can help?  (@TomAugspurger , is there anyone in Anaconda who can help here?)

**Comment by mrocklin** [comment]
@jacobtomlinson can you look at the windows CI failures here?

**Comment by jrbourbeau** [comment]
Yeah, I apologize, the current unfortunate state of things is my fault. I should have had PRs dropping 3.5 in both `dask/dask` and here ready to go concurrently instead of doing them one at a time.

My preference is to wait until the maintainer call tomorrow (maybe someone with TLS experience who has been pinged can investigate between now and then) and discuss this on the call. If nobody's able to debug, we mark the failing TLS tests with `xfail` on Windows, merge this, and open an issue.

**Comment by mrocklin** [comment]
OK, that sounds like a good plan to me.  Thanks James

File: distributed/tests/test_security.py [hunk 1]
```diff
@@ -196,6 +196,9 @@ def many_ciphers(ctx):
     basic_checks(ctx)
     if sys.version_info >= (3, 6):
         supported_ciphers = ctx.get_ciphers()
+        from pprint import pprint
+
+        pprint(supported_ciphers, stream=sys.stderr)
         tls_12_ciphers = [c for c in supported_ciphers if c["protocol"] == "TLSv1.2"]
```
[the remark this answers is not in the document]
Comment (jrbourbeau) [reply distributed/tests/test_security.py]: That seems like a very reasonable change. Thanks for the suggestions @jacobtomlinson!

**Pull request merged**